libs/CovidDatasets.py

```diff
@@ -277,26 +278,40 @@ def parse_country(country):
             return country
 
         day_reports = []
-        for f in os.listdir(daily_reports_dir):
-            if os.path.splitext(f)[1] == '.csv':
-                # For each data file in the directory
-                df = pd.read_csv(os.path.join(daily_reports_dir, f))
-                df = df.rename(columns=self._fieldname_map)
-                df = df.assign(**{self.DATE_FIELD: datetime.datetime.strptime(f.split('.')[0], '%m-%d-%Y')})
-                # Select out the subset of fields we care about
-                if self.COUNTY_FIELD not in df.columns:
-                    df[self.COUNTY_FIELD] = df[self.STATE_FIELD].dropna().apply(parse_county)
-                df = df[[self.DATE_FIELD, self.COUNTRY_FIELD, self.STATE_FIELD, self.COUNTY_FIELD, self.CASE_FIELD,
-                         self.DEATH_FIELD, self.RECOVERED_FIELD]]
-                # Parse the states from their longhand into their abbreviations
-                df[self.STATE_FIELD] = df[self.STATE_FIELD].dropna().apply(parse_state)
-                # Parse the 'US' entries into 'USA'
-                df[self.COUNTRY_FIELD] = df[self.COUNTRY_FIELD].apply(parse_country)
-                # Append the record dates by converting the file name into a datetime object
-                # Only process the csv files
-                day_reports.append(df)
+        snapshot_date = self._FIRST_JHU_DATE
+        while True:
+            csv_url = self._JHU_URL_TEMPLATE.format(snapshot_date.strftime('%m-%d-%Y'))
+            logging.info('Loading URL: {}'.format(csv_url))
+
+            # For each data file
+            try:
+                df = pd.read_csv(csv_url)
+            except urllib.error.HTTPError as e:
+                if e.code == 404:
+                    logging.info('Received a 404 for date {}. Ending iteration.'.format(snapshot_date))
+                    break
+                raise
```

Comment:
I attempted to use the GitHub content API, but I was having issues getting the listing for the folder. Because this only ends on 404, other errors should still get raised.

Reply:
Sounds good.

I wonder a bit if fetching from raw.githubusercontent.com is the best option long-term.  Besides this enumeration problem, it may be a bit slow. A couple potential alternatives:
* Have a shell script or similar that clones the covid-data-public repo (at a revision) and then points run.py and other scripts at that location. Then we can do (fast) local filesystem access.
* We could _consider_ embedding the covid-data-public repo in covid-data-model as a git submodule, but TBH I don't know how suitable that is and the learning curve on git submodules may make it a subpar solution, even if it would work.

But this seems fine to me for now.